igvc/src/pathplanner/main.cpp

```diff
@@ -106,25 +106,37 @@ int main(int argc, char** argv)
 
   double baseline = 0.93;
 
+  ros::NodeHandle pNh("~");
+
   search_problem.Map = pcl::PointCloud<pcl::PointXYZ>().makeShared();
   search_problem.Map->header.frame_id = "/odom";
   search_problem.Octree = boost::make_shared<pcl::octree::OctreePointCloudSearch<pcl::PointXYZ>>(0.1);
   search_problem.Octree->setInputCloud(search_problem.Map);
-  search_problem.GoalThreshold = 1.0;
-  search_problem.Threshold = 0.5;
-  search_problem.Speed = 1.0;
-  search_problem.Baseline = baseline;
+
+  if(!pNh.hasParam("goal_threshold") || !pNh.hasParam("threshold") || !pNh.hasParam("speed") || !pNh.hasParam("baseline") || !pNh.hasParam("minimum_omega") || !pNh.hasParam("maximum_omega") || !pNh.hasParam("delta_omega") || !pNh.hasParam("point_turns_enabled") || !pNh.hasParam("reverse_enabled") || !pNh.hasParam("max_obstacle_delta_t") || !pNh.hasParam("alpha") || !pNh.hasParam("beta") || !pNh.hasParam("bounding_distance")) {
+    ROS_ERROR_STREAM("path planner does not have all required parameters");
+    ROS_ERROR_STREAM("path planner does not have all required parameters");
+    ROS_ERROR_STREAM("path planner does not have all required parameters");
```

Comment:
Duplicate error messages?

Reply:
I should do a ros rate that continues to publish or set default values.

Reply:
It's also OK to just have the node shut itself down if it's started without all of the parameters it needs.

Reply:
That’s actually a much better idea. I will do that.